tests/config/pubsub/tests.yml

```diff
@@ -15,8 +17,12 @@ components:
       testTopicName: dapr-conf-test
   - component: redis
     allOperations: true
+    config:
+      checkInOrderProcessing: false
+      simulateErrors: false
```

Comment:
Why redis cannot simulate errors?

Reply:
Currently Redis does not have the retry logic that Kafka and Pulsar have. It also fires off each message in a separate gorouting which breaks in-order processing. I didn't want to make this PR too big by making significant changes to the Redis component without careful consideration (as an example maybe there be a setting for parallel processing/more than 1 in-flight message). These changes are likely non-trivial and IMO should be review independently.

My thought here was to add the simulate errors flag to allow the conformance test to pass for now. I think once we get all of the pubsub components added and conforming to the rules we envision, this setting can be removed. I'll add a comment to that affect. This is a baby step in thinking about the behaviors and settings of the pubsub components more holistically. Sorry, This was a late finding and I should have commented on the rationale as opposed to just relying on the commit message.

Reply:
So, the Redis implementation cannot handle errors? That is surprising. I can see why it does not support oder in case of error of a single message but cannot recover from an error is another statement.

Reply:
I didn't say it couldn't handle errors, just that we did't have retries there yet. :)
Currently, in case of an error, the message is not Ack'ed and I believe Redis attempts to redeliver it in the future. From what I observed, it is not always redelivered within the timeframe of our conformance test (which is bumped up to 60s in this PR). Allowing for redelivery of course also breaks in-order processing but that will be made configurable in an upcoming issue+PR.

Reply:
Please, keep redis in the test for errors and remove this configuration. It was passing consistently before this change. It might have to do with the extra errors added here. Making this configurable means that components might choose not handle message delivery and still be conformant.

Reply:
Redis was passing because the conformance test did not have the granular message tracking check that this PR introduced so it passing before was a false positive. I went ahead and enhanced the Redis component to support redelivery (reclaiming messages) which makes the test pass without the need of the temporary flag (which I also removed).

Reply:
Outstanding. I will take a look at this again soon.